service/history/executor_factory.go

```diff
@@ -0,0 +1,295 @@
+	workflowCache wcache.Cache,
+	logger log.Logger,
+) queues.Executor {
+	return NewArchivalQueueTaskExecutor(
+		f.Archiver,
+		shardCtx,
+		workflowCache,
+		f.RelocatableAttributesFetcher,
+		f.MetricsHandler,
+		logger,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTimerActiveExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	workflowDeleteManager deletemanager.DeleteManager,
+	logger log.Logger,
+) queues.Executor {
+	return newTimerQueueActiveTaskExecutor(
+		shardCtx,
+		workflowCache,
+		workflowDeleteManager,
+		logger,
+		f.MetricsHandler,
+		f.Config,
+		f.MatchingClient,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTimerStandbyExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	workflowDeleteManager deletemanager.DeleteManager,
+	nDCHistoryResender xdc.NDCHistoryResender,
+	logger log.Logger,
+	clusterName string,
+) queues.Executor {
+	return newTimerQueueStandbyTaskExecutor(
+		shardCtx,
+		workflowCache,
+		workflowDeleteManager,
+		nDCHistoryResender,
+		f.MatchingClient,
+		logger,
+		f.MetricsHandler,
+		clusterName,
+		f.Config,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTransferActiveExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	logger log.Logger,
+) queues.Executor {
+	return newTransferQueueActiveTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.ArchivalClient,
+		f.SdkClientFactory,
+		logger,
+		f.MetricsHandler,
+		f.Config,
+		f.MatchingClient,
+		f.VisibilityManager,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateTransferStandbyExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	nDCHistoryResender xdc.NDCHistoryResender,
+	logger log.Logger,
+	clusterName string,
+) queues.Executor {
+	return newTransferQueueStandbyTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.ArchivalClient,
+		nDCHistoryResender,
+		logger,
+		f.MetricsHandler,
+		clusterName,
+		f.MatchingClient,
+		f.VisibilityManager,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateExecutorWrapper(
+	currentClusterName string,
+	activeExecutor queues.Executor,
+	standbyExecutor queues.Executor,
+	logger log.Logger,
+) queues.Executor {
+	return queues.NewExecutorWrapper(
+		currentClusterName,
+		f.NamespaceRegistry,
+		activeExecutor,
+		standbyExecutor,
+		logger,
+	)
+}
+
+func (f *ExecutorFactoryBase) CreateVisibilityExecutor(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+	logger log.Logger,
+) queues.Executor {
+	return newVisibilityQueueTaskExecutor(
+		shardCtx,
+		workflowCache,
+		f.VisibilityManager,
+		logger,
+		f.MetricsHandler,
+		f.Config.VisibilityProcessorEnsureCloseBeforeDelete,
+		f.Config.VisibilityProcessorEnableCloseWorkflowCleanup,
+	)
+}
+
+func (f *ExecutorFactoryBase) GetNewWorkflowDeleteManager(
+	shardCtx shard.Context,
+	workflowCache wcache.Cache,
+) deletemanager.DeleteManager {
+	return deletemanager.NewDeleteManager(
+		shardCtx,
+		workflowCache,
+		f.Config,
+		f.ArchivalClient,
+		shardCtx.GetTimeSource(),
+		f.VisibilityManager,
+	)
+}
```

Comment:
This function isn't called anywhere, and it looks like it's duplicated in the timer queue factory here: `func (f *timerQueueFactory) getNewDeleteManager(shardCtx shard.Context, workflowCache wcache.Cache) deletemanager.DeleteManager`. Also, why does this belong in the executor factory? It's not building an executor.

Reply:
oops, meant to remove this